### PR TITLE
feat: search in contacts #2010

### LINKED ISSUE #2010: Search in contacts

### PR DESCRIPTION
**Pull request by frolyo**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Services/Search/SearchUI.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using ActualLab.Interception;
+
+namespace ActualChat.Chat.UI.Blazor.Services;
+
+public partial class SearchUI : ScopedWorkerBase<ChatUIHub>, IComputeService, INotifyInitialized
+{
+    private Cached _contactSearchResults = Cached.None;
+
+    public MutableState<string> Text { get; }
+    public MutableState<bool> IsSearchModeOn { get; }
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Services/Search/SearchUI.cs]: better to expose it as IState<bool>

File: src/dotnet/Core/DelegatingWorker.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        await base.Run().ConfigureAwait(false);
+        return Result!;
+    }
+}
+
+public sealed class DelegatingWorker(
+    Func<CancellationToken, Task> jobFactory,
+    CancellationTokenSource? stopTokenSource = null,
+    bool start = true) : DelegatingWorker<Unit>(jobFactory.ToUnitTaskFactory(), stopTokenSource)
```
Comment (crui3er) [inline src/dotnet/Core/DelegatingWorker.cs]: start parameter is not used, remove it

**Review by crui3er — DISMISSED** [review]

File: src/dotnet/Chat.UI.Blazor/Services/Search/SearchUI.cs (hunk 1 above)
Comment (frolyo) [reply src/dotnet/Chat.UI.Blazor/Services/Search/SearchUI.cs]: done

File: src/dotnet/Core/DelegatingWorker.cs (hunk 2 above)
Comment (frolyo) [reply src/dotnet/Core/DelegatingWorker.cs]: done

**Review by alexis-kochetov — APPROVED** [review]

**Pull request merged**